server/src/test/java/org/elasticsearch/transport/RemoteClusterServiceTests.java

```diff
@@ -66,14 +67,30 @@
 public class RemoteClusterServiceTests extends ESTestCase {
 
     private final ThreadPool threadPool = new TestThreadPool(getClass().getName());
+    private LinkedProjectConfigService linkedProjectConfigService = null;
 
     @Override
     public void tearDown() throws Exception {
         super.tearDown();
         ThreadPool.terminate(threadPool, 10, TimeUnit.SECONDS);
     }
 
-    private RemoteClusterService createRemoteClusterService(final Settings settings, final MockTransportService transportService) {
+    private RemoteClusterService createRemoteClusterService(Settings settings, MockTransportService transportService) {
+        return createRemoteClusterService(settings, ClusterSettings.createBuiltInClusterSettings(), transportService);
+    }
+
+    private RemoteClusterService createRemoteClusterService(
+        Settings settings,
+        ClusterSettings clusterSettings,
+        MockTransportService transportService
+    ) {
+        if (linkedProjectConfigService == null) {
+            linkedProjectConfigService = new ClusterSettingsLinkedProjectConfigService(
+                settings,
+                clusterSettings,
+                DefaultProjectResolver.INSTANCE
+            );
+        }
```

Comment:
Do we need this conditional creation? It can potentially make test change in one method affect subsequent test methods. Maybe you have checked the usages and this is not an issue?